onnxruntime/test/contrib_ops/matmul_nbits.cc

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ORT_MINIMAL_BUILD
+
+#include "core/common/span_utils.h"
+#include "core/framework/tensor.h"
+#include "core/mlas/inc/mlas_qnbit.h"
+#include "core/session/inference_session.h"
+#include "test/common/tensor_op_test_utils.h"
+#include "test/framework/test_utils.h"
+#include "test/optimizer/graph_transform_test_builder.h"
+#include "test/providers/provider_test_utils.h"
+#include "test/util/include/default_providers.h"
+#include "core/util/qmath.h"
+
+#include <chrono>
+#include <random>
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+namespace onnxruntime {
+namespace test {
+
+void MlasJblasQ4Test(int64_t M, int64_t N, int64_t K, int block_size, bool is_asym, MLAS_SQNBIT_COMPUTE_TYPE acc_lvl,
```

Comment:
It appears that the tests are associated with `onnxruntime/test/contrib_ops/matmul_4bits_test.cc`. Could you please relocate the tests to this file and rename them to SharedPrepackedWeights? These tests actually evaluate SharedPrepackedWeights, not just MlasJblasQ4Test.